vulture_os/toolkit/api_parser/sentinel_one_identity_graph/sentinel_one_identity_graph.py

```diff
@@ -0,0 +1,263 @@
+            fragment Indicators on Indicator {attacks {...Attacks} eventTime message observables {...Observables} severity type}
+            fragment RealTime on RealTime {scope {...Scope}}
+            query GetAlert($id:ID!) {
+                alert(id:$id) {
+                    analystVerdict
+                    analytics {...Analytics}
+                    asset {...Asset}
+                    assignee {...Assignee}
+                    attackSurfaces
+                    classification
+                    confidenceLevel
+                    createdAt
+                    description
+                    detectedAt
+                    detectionSource {...DetectionSource}
+                    detectionTime {...DetectionTime}
+                    externalId
+                    fileHash
+                    fileName
+                    firstSeenAt
+                    id
+                    indicators {...Indicators}
+                    lastSeenAt
+                    name
+                    noteExists
+                    process {...Process}
+                    realTime {...RealTime}
+                    result
+                    severity
+                    status
+                    storylineId
+                    updatedAt
+                }
+            }
+        """
+
+        response = self.execute_query(
+            self.url,
+            data={'query': query, 'variables': {'id': event_id}}
+        )
+
+        return response.get('data', {}).get('alert', {})
+
+    def format_log(self, log: dict) -> str:
+        return dumps(log)
+
+    def test(self):
+        logs = []
+
+        try:
+            since = timezone.now() - timedelta(days=1)
+            to = timezone.now()
+
+            alert_ids = self.get_itdr_events_ids(
+                since=int(since.timestamp()*1000),
+                to=int(to.timestamp()*1000),
+                limit=100)
+
+            for id in alert_ids:
+                logs.append(self.get_itdr_event_details(id))
+
+            return {
+                "status": True,
+                "data": logs
+            }
+        except Exception as e:
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def execute(self):
+        kind = "sentinel_one_identity_graph_alerts"
+
+        current_time = timezone.now()
+
+        since = datetime.strptime(self.last_collected_timestamps.get(kind), "%Y-%m-%dT%H:%M:%SZ") or current_time - timedelta(days=7)
+        to = min(since + timedelta(days=1), current_time - timedelta(minutes=3))
+
+        while to < current_time - timedelta(minutes=3):
+            since = datetime.strptime(self.last_collected_timestamps.get(kind), "%Y-%m-%dT%H:%M:%SZ") or timezone.now() - timedelta(days=7)
+            to = min(since + timedelta(days=1), current_time - timedelta(minutes=3))
```

Comment:
- avoid updating the `since` and `to` limits during processing (each run every minute is enough to catch up progressively)
- don't forget to take into account the `evt_stop` if possible, and if collection can take a lot of time (handle partial collection if handled)